sdk/monitor/monitor-query/README.md

```diff
@@ -0,0 +1,174 @@
+
+```javascript
+const { DefaultAzureCredential } = require("@azure/identity");
+const { LogsClient, MetricsClient } = require("@azure/monitor-query");
+
+const credential = new DefaultAzureCredential();
+
+const logsClient = new LogsClient(credential);
+// or
+const metricsClient = new MetricsClient(credential);
+```
+
+More information about `@azure/identity` can be found [here](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/README.md)
+
+## Key concepts
+
+The [`LogsClient`]<!--[msdocs_logs_client]--> allows you to query logs, using the [Kusto query language][kusto_query_language]. This data can be queried in the
+portal using tables like `AppEvents`, `AppDependencies` and others.
+
+The [`MetricsClient`]<!--[msdocs_metrics_client]--> allows you to query metrics.
+
+## Examples
+
+#### Querying logs
+
+The LogsClient can be used to query a Monitor workspace using the Kusto Query language.
+
+```javascript
+const monitorQuery = require("@azure/monitor-query");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+const azureLogAnalyticsWorkspaceId = "<the Workspace Id for your Azure Log Analytics resource>";
+const logsClient = new monitorQuery.LogsClient(new DefaultAzureCredential());
+
+async function run() {
+  const kustoQuery = "AppEvents | limit 1";
+  const result = logsClient.queryLogs(azureLogAnalyticsWorkspaceId, kustoQuery);
+  const tablesFromResult: Table[] | undefined = result.tables;
+
+  if (tablesFromResult == null) {
+    console.log(`No results for query '${kustoQuery}'`);
+    return;
+  }
+
+  console.log(`Results for query '${kustoQuery}'`);
+
+  for (const table of tablesFromResult) {
+    const columnHeaderString = table.columns
+      .map((column) => `${column.name}(${column.type}) `)
+      .join("| ");
+    console.log("| " + columnHeaderString);
+
+    for (const row of table.rows) {
+      const columnValuesString = row.map((columnValue) => `'${columnValue}' `).join("| ");
+      console.log("| " + columnValuesString);
+    }
+  }
+}
+run().catch((err) => console.log("ERROR:", err));
+```
+
+For more samples see here: [samples][samples].
+
+## Troubleshooting
+
+### Logging
+
+Enabling logging may help uncover useful information about failures. In order to see a log of HTTP requests and responses, set the `AZURE_LOG_LEVEL` environment variable to `info`. Alternatively, logging can be enabled at runtime by calling `setLogLevel` in the `@azure/logger`:
+
+```javascript
+import { setLogLevel } from "@azure/logger";
+
+setLogLevel("info");
+```
+
+For more detailed instructions on how to enable logs, you can look at the [@azure/logger package docs](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/core/logger).
+
+## Next steps
+
+The following samples show you the various ways you can query your Log Analytics workspace:
+
+- [`logsQuery.ts`]<!--[samples_logsquery_ts]--> - Query logs in a Monitor workspace
```

Comment:
should these be working links?

Reply:
Not yet (since we haven't checked in a first rev of this code into main). 

I'll basically do a commit right after this one to turn all of these links on.